internals/secrethub/account_delete.go

```diff
@@ -0,0 +1,91 @@
+package secrethub
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/secrethub/secrethub-cli/internals/cli"
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/iterator"
+)
+
+// AccountDeleteCommand is a command to inspect account details.
+type AccountDeleteCommand struct {
+	io        ui.IO
+	newClient newClientFunc
+}
+
+// NewAccountDeleteCommand creates a new AccountDeleteCommand.
+func NewAccountDeleteCommand(io ui.IO, newClient newClientFunc) *AccountDeleteCommand {
+	return &AccountDeleteCommand{
+		io:        io,
+		newClient: newClient,
+	}
+}
+
+// Register registers the command, arguments and flags on the provided Registerer.
+func (cmd *AccountDeleteCommand) Register(r cli.Registerer) {
+	clause := r.Command("delete", "Delete your SecretHub account.")
+
+	clause.BindAction(cmd.Run)
+}
+
+// Run handles the command with the options as specified in the command.
+func (cmd *AccountDeleteCommand) Run() error {
+	client, err := cmd.newClient()
+	if err != nil {
+		return err
+	}
+	var account *api.Account
+	account, err = client.Accounts().Me()
+	if err != nil {
+		return err
+	}
+
+	var orgList []api.Org
+	orgIterator := client.Orgs().Iterator(nil)
+	for {
+		var org api.Org
+		org, err = orgIterator.Next()
+		if err == iterator.Done {
+			break
+		} else if err != nil {
+			return err
+		}
+		orgList = append(orgList, org)
+	}
+	if len(orgList) > 0 {
+		builder := strings.Builder{}
+		for i, org := range orgList {
+			builder.WriteString(org.Name)
+			if i != len(orgList)-1 {
+				builder.WriteString(", ")
+			}
+		}
+		return fmt.Errorf("cannot delete account that is a member of an organization. Please leave or delete the following organizations before deleting your account: %s", builder.String())
```

Comment:
To make this more user-friendly we should do one of two things:
1. Make it possible to interactively leave or delete orgs from the `account delete` command.
2. Provide clearer instructions with the exact commands to execute to leave or delete orgs.

Reply:
Shall I implement an interactive "wizard" that handles this? Something like:
```
if user is not last org admin {
    leave org
} else if user is last member of org {
    delete org
} else  {
    choose someone else to be admin instead of you
}
```

This would be done for each of the orgs that you are a member of, and, of course it would prompt for confirmation (type the org name to confirm, etc.).

Does this look good to you?

Reply:
Yep, that looks about right :+1:

I'm wondering if removing the org should be an option in all scenarios:
```
if user is last member of org {
    prompt with confirmation to remove the org
} else if other admin exists {
    prompt with question to leave or remove the org
} else {
    prompt with question to promote someone to admin or remove the org
}
```

Since depending on what scenario you're in for removing your account, there may not be a use case for handing over the org anymore. For example: if you're an admin of an org that you've migrated to 1Password, you probably don't want to promote someone else to the admin role, but just want to remove the data as now 1Password is the source of truth for it.

What do you think?

Reply:
Yupp :+1:  I had the same idea but couldn't come up with a use case for it. Migrating to 1Password sounds like a good use case to support with this prompt.

Reply:
I slightly deviated from this plan but I think the result makes more sense.